### PR TITLE
refactor(database/offline): drop tables after row read complete

### PR DESCRIPTION
**Pull request by jinghancc**
This PR fixes offline store method `Join`: drop temporary tables after row read complete.

### CODE REVIEW

**Review by YikSanChan — APPROVED** [review]
lgtm

File: internal/database/offline/postgres/join.go [hunk 1]
```diff
@@ -134,9 +134,14 @@ func (db *DB) readJoinedTable(ctx context.Context, entityRowsTableName string, t
...
+	var scanErr, dropErr error
 	go func() {
-		defer rows.Close()
+		defer func() {
+			if err := db.dropTemporaryTables(ctx, tableNames); err != nil {
+				dropErr = err
+			}
+			rows.Close()
+		}()
 		defer close(data)
```
Comment (wfxr) [inline internal/database/offline/postgres/join.go:145]: Move `close(data)` into `defer func() {...}` ?

**Review by wfxr — APPROVED** [review]
LGTM

File: internal/database/offline/postgres/join.go (hunk 1 above)
Comment (jinghancc) [reply internal/database/offline/postgres/join.go]: fixed

**Review by lianxmfor — APPROVED** [review]
lgtm

**Pull request merged**